code/modules/mob/living/simple_mob/subtypes/animal/alien animals/teppi.dm

```diff
@@ -0,0 +1,1195 @@
+		to_chat(src, "<span class='notice'>I'm not old enough to make babies.</span>")
+		return
+	if(baby_countdown > 0)
+		to_chat(src, "<span class='notice'>It is not time yet...</span>")
+		return
+	if(!breedable || nutrition < 500)
+		to_chat(src, "<span class='notice'>The conditions are not right to produce offspring.</span>")
+		return	
+	if(GLOB.teppi_count >= GLOB.max_teppi) //if we can't make more then we shouldn't look for partners
+		to_chat(src, "<span class='notice'>I cannot produce more offspring at the moment, there are too many of us!</span>")
+		return
+	. = FALSE
+	for(var/mob/living/simple_mob/vore/alienanimals/teppi/alltep in oview(1,src))
+		if(!alltep.teppi_adult || alltep.nutrition < 250 || alltep.prevent_breeding)
+			continue
+		if(alltep)
+			log_admin("[key_name_admin(src)] produced a baby teppi at [get_area(src)] - [COORD(src)]") //Won't show up in the chat, but makes a log of who's having babies where, for investigative purposes.
+			new /mob/living/simple_mob/vore/alienanimals/teppi/baby(loc, src, alltep)
+			baby_countdown = 400 //You don't have a random chance to deal with so the cooldown is twice as long.
+			if(affinity[alltep.real_name])
+				return
+			handle_affinity(alltep, 30) //Mom and dad should like eachother when they do their business
+			alltep.handle_affinity(src, 30)
+			return
+	if(. == FALSE)
+		to_chat(src, "<span class='notice'>There are no suitable partners nearby.</span>")
+
+/mob/living/simple_mob/vore/alienanimals/teppi/proc/toggle_producing_offspring()
+	set name = "Toggle Producing Offspring"
+	set category = "Abilities"
+	set desc = "You can toggle whether or not you can produce offspring."
+	if(!prevent_breeding)
+		to_chat(src, "<span class='notice'>You disable breeding.</span>")
+		prevent_breeding = TRUE
+	else
+		to_chat(src, "<span class='notice'>You enable breeding.</span>")
+		prevent_breeding = FALSE
+		
+///////////////////AI Things////////////////////////
+//Thank you very much Aronai <3
+
+/mob/living/simple_mob/vore/alienanimals/teppi/proc/do_I_know_you()
+	// Get list of everyone who can see us (which is everyone we can see, typically)
+	var/list/people_nearby = oviewers(world.view, src)
+	// Use the hidden . var to avoid needing to create a new local var (saves CPU)
+	. = list()
+	// Add everyone nearby to the list if they're in affinity, with key of the mob and value of the affinity
+	// . becomes list(jane = 1, tim = -3) etc
+	for(var/mob/living/M in people_nearby)
+		var/their_affinity = affinity[M.real_name]
+		if(their_affinity)
+			if(their_affinity >= 25 || their_affinity <= -10)
+				.[M] = affinity[M.real_name]
+	// Sort the list (timsort default sort comperator is numeric ascending, so highest affinity will be last in the list)
+	sortTim(., associative = TRUE)
+
+/datum/ai_holder/simple_mob/teppi/handle_wander_movement()
+	var/mob/living/simple_mob/vore/alienanimals/teppi/tepholder = holder
+	if(tepholder.resting)
+		if(prob(5))
+			tepholder.lay_down()
```

Comment:
Suggestion: Could be a single if()

Reply:
If I do this as a single if then the teppi wanders while it's resting. This tells it to return instead of wandering when it's resting, but also check to see if it's time to wake up.

Reply:
oic, I misread